web/unit_tests/cases/engine.js

```diff
@@ -240,3 +223,82 @@ describe('Unmatched Final Groups', function() {
     runKeyboardTestFromJSON('/engine_tests/ghp_enter.json', {usingOSK: true}, done, assert.equal, testconfig.timeouts.scriptLoad);
   });
 });
+
+// Kept separate to maintain an extra-clean setup for this test.
+describe('Engine - Browser Interactions', function() {
+  this.timeout(testconfig.timeouts.scriptLoad);
+
+  before(function() {
+    fixture.setBase('fixtures');
+  });
+
+  beforeEach(function() {
+    fixture.load("singleInput.html");
+    return setupKMW(null, testconfig.timeouts.scriptLoad);
+  });
+
+  afterEach(function() {
+    fixture.cleanup();
+    teardownKMW();
+  });
+
+  describe('Keyboard Loading', function() {
+    it('Local', function() {
+      this.timeout(testconfig.timeouts.scriptLoad);
+
+      var test_callback = function() {
+        assert.isNotNull(keyman.getKeyboard("lao_2008_basic", "lo"), "Keyboard stub was not registered!");
+        assert.equal(keyman.getActiveKeyboard(), "Keyboard_lao_2008_basic", "Keyboard not set automatically!");
+        keyman.removeKeyboards('lao_2008_basic');
+        assert.equal(keyman.getActiveKeyboard(), '', "Keyboard not removed correctly!");
+      }
+
+      return loadKeyboardFromJSON("/keyboards/lao_2008_basic.json", testconfig.timeouts.scriptLoad)
+        .then(test_callback);
+    });
+
+    it('Automatically sets first available keyboard', function() {
```

Comment:
This unit test covers the feature underlying the instability reported by #5799 and ensures we can stably test it now.